sys-apps/nawk: remove old, fix HOMEPAGE and SRC_URI

mm1ke opened the pull request:
Hi,

This PR removes the old r2 ebuild.
The r3 ebuild is at EAPI6, has no open bugs and is since 262 days in tree.

Further more I've updated HOMEPAGE and the SRC_URI. The Homepage is also used by Arch.

Please review.

junghans commented on sys-apps/nawk/nawk-20121220-r3.ebuild:
```diff
@@ -1,13 +1,13 @@
-HOMEPAGE="http://cm.bell-labs.com/cm/cs/awkbook/index.html"
-SRC_URI="mirror://gentoo/${P}.tar.gz"
+HOMEPAGE="https://www.cs.princeton.edu/~bwk/btl.mirror/"
+SRC_URI="https://www.cs.princeton.edu/~bwk/btl.mirror/awk.tar.gz"
```
Please keep the current `SRC_URI`, `awk.tar.gz` is unversioned and that leads to problems with mirroring.

mm1ke commented:
Hi, I've changed the ebuild in order to rename the downloaded file. Hope this helps?
Please review.

junghans commented on sys-apps/nawk/nawk-20121220-r3.ebuild:
```diff
@@ -1,13 +1,13 @@
-HOMEPAGE="http://cm.bell-labs.com/cm/cs/awkbook/index.html"
-SRC_URI="mirror://gentoo/${P}.tar.gz"
+HOMEPAGE="https://www.cs.princeton.edu/~bwk/btl.mirror/"
+SRC_URI="https://www.cs.princeton.edu/~bwk/btl.mirror/awk.tar.gz -> ${P}.tar.gz"
```
No that doesn't work either, as upstream unpredictably changes the content of the tarball and then if would be a different version.

mm1ke replied:
Hmm.. yes i see.
Well, i guess my options are now to either i ask upstream to provide a versioned package (which, considering the state of the package, won't happen) or stay with the gentoo mirror for the source?

junghans replied:
It never hurt to ask though...

I also email them that patch https://github.com/junghans/nawk/commit/1ab0265fec229c5e15d2f252360eca1834c679c7 at some point.

mm1ke replied:
Sure, I'll send a mail and ask for a versioned package.
Can we keep the PR open in the mean time?

mm1ke commented:
Hi,

Sadly i haven't got any response. I've rebased and updated this PR again to use the gentoo mirror sources. Should i get a answer someday i still can make a new PR :)
